nupic/data/record_stream.py

```diff
@@ -25,178 +25,271 @@
 import datetime
 
 
+from nupic.data.fieldmeta import FieldMetaSpecial
 
-class RecordStreamIface(object):
-  """This is the interface for the record input/output storage classes."""
 
-  __metaclass__ = ABCMeta
 
+def _getFieldIndexBySpecial(fields, special):
+  """ Return index of the field matching the field meta special value.
+  :param fields: sequence of nupic.data.fieldmeta.FieldMetaInfo objects
+    representing the fields of a stream
+  :param special: one of the special field attribute values from
+    nupic.data.fieldmeta.FieldMetaSpecial
+  :returns: first zero-based index of the field tagged with the target field
+    meta special attribute; None if no such field
+  """
+  for i, field in enumerate(fields):
```

Comment:
Maybe a good idea to check that `fields` is in fact a sequence that can be indexed (like a `list` or `tuple`).  A subtle bug could be introduced if someone accidentally passes a `dict` or `set`.